### PR TITLE
#7534 changes to the buildConfig calls

### PR DESCRIPTION
**Pull request by MV88**
## Description

changes to the buildConfig calls

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [x] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7534

**What is the new behavior?**

it uses object for webpack.config and prod.webpack.config which makes things clearer

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: build/prod-webpack.config.js [hunk 1]
```diff
@@ -21,11 +21,11 @@ module.exports = require('./buildConfig')(
...
     paths,
-    [extractThemesPlugin, ModuleFederationPlugin],
-    true,
-    undefined,
-    undefined,
-    [
+    plugins: [extractThemesPlugin, ModuleFederationPlugin],
+    prod: true,
+    publicPath: undefined,
+    cssPrefix: undefined,
```
Comment (offtherailz) [inline build/prod-webpack.config.js:27]: you can remove these I think

**Review by offtherailz — COMMENTED** [review]
LGTM

File: build/prod-webpack.config.js (hunk 1 above)
Comment (MV88) [reply build/prod-webpack.config.js:27]: yeah right
i was in doubt if we had to make them clearly specified and visible

Comment (MV88) [reply build/prod-webpack.config.js:27]: can you make the suggestion and remove it?

Comment (MV88) [inline build/prod-webpack.config.js:27]:
```suggestion
```

**Review by offtherailz — APPROVED** [review]

**Pull request merged**